h/views/api/annotations.py

```diff
@@ -166,6 +166,28 @@ def delete(context, request):
     return {"id": context.annotation.id, "deleted": True}
 
 
+@api_config(
+    versions=["v1", "v2"],
+    route_name="api.annotation.reindex",
+    request_method="POST",
+    permission=Permission.Annotation.UPDATE,
+    link_name="annotation.reindex",
+    description="Reindex an annotation",
+)
+def reindex(context, request):  # pragma: no cover
```

Comment:
This still needs tests and a mechanism to disable it on prod.

For the latter I'm leaning to just add and explicit conditional in the view body.